lebedigital/calibration/utils.py

```diff
@@ -0,0 +1,195 @@
+sys.path.append(os.path.join(baseDir1, 'Data'))
+
+#import emodul_query
+
+def query_KG(path: str, skip_last = 145, skip_init = 330 ) -> dict:
+    """
+    Queries KG, the updated one done by Illias (I think from KG team). Lot of funky words as I dont know what they are.
+    Returns:
+    Args:
+        path ():
+        skip_last (): Last datavalues to skip as we only need the third loading cycle.
+        skip_init (): Initial datavalues to skip as we only need the third loading cycle.
+
+    Returns:
+
+    """
+    """
+    Queries KG, the updated one done by Illias (I think from KG team). Lot of funky words as I dont know what they are.
+    Returns:
+
+    """
+
+    def query_objects(queries, graph):
+        # function to get objects from specific subject, predicate pairs, knowing there is only one result
+        results = {}
+        for query in queries:
+            q = f"""
+                            SELECT ?object
+                            WHERE {{
+                                    {queries[query]['subject']}  {queries[query]['predicate']} ?object
+                            }}
+                    """
+
+            result = graph.query(q)
+            for r in result:
+                results[query] = r["object"]
+
+        return results
+
+    # define queries
+    queries = {
+        'diameter': {'subject': 'ns1:informationbearingentity1', 'predicate': 'ns1:has_decimal_value'},
+        'length': {'subject': 'ns1:informationbearingentity2', 'predicate': 'ns1:has_decimal_value'},
+        'path': {'subject': 'ns1:informationbearingentity9', 'predicate': 'ns9:has_text_value'},
+        'file_name': {'subject': 'ns1:informationbearingentity8', 'predicate': 'ns9:has_text_value'}
+    }
+
+    # Path to KG, for testing
+    #path_to_KG = '../../usecases/MinimumWorkingExample/emodul/knowledge_graphs/BA-Losert MI E-Modul 28d v. 04.08.14 Probe 4.ttl'
+    path_to_KG = path
+    # initialize the graph
+    knowledge_graph = rdflib.Graph()
+    knowledge_graph.parse(path_to_KG, format='ttl')
+
+    # queries
+    results = query_objects(queries, knowledge_graph)
+
+    # read csv into dataframe
+    file_path = results['path'] + '/' + results['file_name']
+    #skip_last = 145 # hard coded to get the third loading cycle. need somthing better
+    #skip_init = 330
+    df = pd.read_csv(file_path, skipfooter=skip_last, engine='python')
+    df = df.drop(labels=range(0, skip_init), axis=0)
+    df['displacement'] = (df['Transducer 1[mm]'] + df['Transducer 2[mm]'] + df['Transducer 3[mm]']) / 3
+    df['stress'] = df['Force [kN]'] / (np.pi * (float(results['diameter']) / 2) ** 2)
+
+    output = {
+        'height': float(results['length']),
+        'diameter': float(results['diameter']),
+        'displacement': np.array(df['displacement']),
+        'stress': np.array(df['stress'])}
+    return output
+
+def load_experimental_data(exp_name, skip_init, skip_last, KG=False, path=None):
```

Comment:
Should we have the scripts that perform some analysis of the data for a specific test go into that directory instead of putting that into calibration?

Reply:
Can you please elaborate what is "that" directory. Didnt get it.

Reply:
In load_experimental_data you load in a file spefically related to a Young's modulus test (this would not work for the hydration test data). How would you name the load_experimental_data for the hydration tests, since this name is already taken? Also, the calibration that you perform is specifically related only to Young's modulus data (the hydration tests would require different inputs, outputs, models, ..). Thus, the suggestion is to move that to a directory related to the corresponding experiment, e.g. calibration/labtests/YoungsModulus.